ath79: add Mikrotik mAP lite support

llogar opened the pull request:
... and fix some minor errors in kernel ar71xx_regs.h include file.

As boot loader leaves some of the gpio pins that control leds in a not-controllable-by-gpio-output state, a DT gpio mask property is introduced that can be used to properly configure the gpio output pins (more or less copy pasted from the cod e from ar71xx target)

llogar committed:
kernel: fix ar71xx registers names and/or addresses

See the ar7xxx and qca953x datasheets.

Signed-off-by: Luka Logar <luka.logar@iname.com>

llogar committed:
kernel: add mask property to DT gpio

Some devices (like Mikrotik mAP lite) need extra gpio initialization in order for the leds
connected to gpios to be controllable via the gpio pins. Setting bit to 1 resets
the gpio output mux for the specified gpio pin.

Signed-off-by: Luka Logar <luka.logar@iname.com>

adschm commented:
Please add Specifications and flashing instructions to your device support patch (just look at other similar patches if you need guidelines ...)

adschm commented on target/linux/ath79/image/mikrotik.mk:
```diff
@@ -62,3 +62,11 @@ define Device/mikrotik_routerboard-wapr-2nd
 endef
 TARGET_DEVICES += mikrotik_routerboard-wapr-2nd
+
+define Device/mikrotik_routerboard-mapl-2nd
```
Please apply proper alphabetic sorting.

llogar replied:
Should I reorder just the `mikrotik.mk` or also `board.d/*` and `hotplug.d/*` files?

adschm replied:
Well, all of them should be sorted alphabetically ...

llogar replied:
Sorted.

adschm commented on target/linux/ath79/mikrotik/base-files/etc/board.d/01_leds:
```diff
@@ -9,6 +9,8 @@ board=$(board_name)
 mikrotik,routerboard-lhg-2nd)
 	ucidef_set_led_netdev "lan" "lan" "green:lan" "eth0"
+mikrotik,routerboard-mapl-2nd)
+	ucidef_set_led_netdev "lan" "lan" "green:eth" "eth0"
```
Missing ";;". Thanks for showing us that this is not tested.

llogar replied:
You are welcome :(
Sorry, I have only one device and it's in use (running OpenWrt), so I can not run-test every change...

adschm replied:
Well, would have saved you from posting a non-running version.

adschm commented on target/linux/ath79/image/mikrotik.mk:
```diff
@@ -62,3 +70,4 @@ define Device/mikrotik_routerboard-wapr-2nd
   IMAGE_SIZE := 16256k
 endef
 TARGET_DEVICES += mikrotik_routerboard-wapr-2nd
+
```
drop empty line at EOF.

llogar replied:
Done.

llogar committed:
ath79: add Mikrotik mAP lite support

Device specification: https://mikrotik.com/product/RBmAPL-2nD

Network over PWR LINE is not supported.
Gpios for the leds have to be initialized using the gpio mask property.

Flash per usual Mikrotik procedure: https://openwrt.org/toh/mikrotik/common
First netboot the initramfs file, then do the sysupgrade ...

Signed-off-by: Luka Logar <luka.logar@iname.com>

CodeFetch commented on target/linux/ath79/dts/qca9533_mikrotik_routerboard-mapl-2nd.dts:
```diff
@@ -0,0 +1,48 @@
+
+/ {
+	compatible = "mikrotik,routerboard-mapl-2nd", "qca,qca9533";
+	model = "MikroTik RouterBOARD mAPL-2nD (mAP lite)";
```
The model string should match the compatible string:
`mikrotik,routerboard-mapl-2nd-map-lite`

CodeFetch replied:
@adschm Or is it okay?

adschm replied:
"mAP lite" is an _alternate_ name, and it would not make sense to include two names into the compatible. So, IMO the current choice is okay. Whether one should choose mAPL-2nD or "mAP lite" for the primary name is subject of ongoing disagreement, though.

Tekagi01 reviewed (approved):
Implemented changes, compiled and everything is working as expected.

ynezz commented:
Merged via https://github.com/openwrt/openwrt/pull/9361
